File: 2485-find-the-pivot-integer/2485-find-the-pivot-integer.cpp
class Solution {
public:
    int sum(int low, int high) {
        int res = 0;
        for (int i = low; i <= high; i++) {
            res += i;
        }
        return res;
    }
    
    int pivotInteger(int n) {
        for (int x = 1; x <= n; x++) {
            if (sum(1, x) == sum(x, n)) {
                return x;
            }
        }
        return -1;
    }
};

File: 2485-find-the-pivot-integer/2485-find-the-pivot-integer.java
class Solution {
    public int pivotInteger(int n) {
        int totalSum = n * (n + 1) / 2;
        int sum = 0;
        for (int i = 1; i <= n; i++) {
            sum += i;
            if (sum == totalSum) {
                return i;
            }
            totalSum -= i;
        }
        return -1;
    }
}